ext/mysql2/extconf.rb

```diff
@@ -125,7 +127,7 @@ def asplode(lib)
 end
 
 usable_flags = wishlist.select do |flag|
-  try_link('int main() {return 0;}', flag)
+  try_link('int main() {return 0;}', "-Werror -Wunknown-warning-option #{flag}")
```

Comment:
Clever! In the event that `-Wunknown-warning-option` itself is unsupported, it should reject all of the flags.